### DISCUSSION TITLE
Query rules from API

### DISCUSSION THREAD

**Discussion by renatomotorline** [lf-edge/ekuiper#779] [Q&A]
Hi,

Is possible to query the rules from the API?
Example: I have an external API with multiple users creating rules, each rule id is "$user_id-$rule_number".
I want to get only the user rules. Is any way to do this?
From the documentation (https://github.com/emqx/kuiper/blob/master/docs/en_US/restapi/rules.md) I understand that is only possible to get all. I can get all and filter the response but if I have many rules this will affect the performance.

**Comment by ngjaying** [comment]
Hi @renatomotorline ,

Currently, query is not supported for rules API.  The only way I can think of is to get all rules and filter as what you already pointed out.

**Reply by renatomotorline** [reply]
Hi @ngjaying ,

It is a feature to implement in the near future or don't have a prediction?

Thanks.

**Reply by ngjaying** [reply]
Hi Renato,

My opionion is NOT to add this feature because the implementation will still need to loop through all rules to filter which has the same performance overhead. And your rule name pattern "$user_id-$rule_number" is actually a private convention which is more appropriate to be handled by yourself.
